aws_google_auth/__init__.py

```diff
@@ -312,18 +327,63 @@ def cli():
     doc = etree.fromstring(base64.b64decode(encoded_saml))
     roles = dict([x.split(',') for x in doc.xpath('//*[@Name = "https://aws.amazon.com/SAML/Attributes/Role"]//text()')])
 
-    role, provider = pick_one(roles)
+    if not config.role_arn in roles:
+        config.role_arn, config.provider = pick_one(roles)
 
-    print "Assuming " + role
+    print "Assuming " + config.role_arn
 
-    sts = boto3.client('sts', region_name=REGION)
+    sts = boto3.client('sts', region_name=config.region)
     token = sts.assume_role_with_saml(
-                RoleArn=role,
-                PrincipalArn=provider,
+                RoleArn=config.role_arn,
+                PrincipalArn=config.provider,
                 SAMLAssertion=encoded_saml,
-                DurationSeconds=args.duration)
+                DurationSeconds=config.duration)
+
+    _store(config, token)
+
+
+def _store(config, aws_session_token):
 
-    print "export AWS_ACCESS_KEY_ID='{}'".format(token['Credentials']['AccessKeyId'])
```

Comment:
If `--profile` is specified, don't do this bit; otherwise, of course, write the profile.